toptal/picasso
# [FX-5602] Migrate PageTopBarMenu to TailwindCSS

[FX-5602]

### Description

This PR migrates `PageTopBarMenu` to TailwindCSS and solves [FX-5955 bug](https://toptal-core.atlassian.net/browse/FX-5955).

### How to test


1. [Temploy](https://picasso.toptal.net/fx-5692-migrate-pagetopbarmenu)
2. Check bug

To check the bug use following code in [temploy](https://picasso.toptal.net/fx-5692-migrate-pagetopbarmenu) and in [picasso](https://picasso.toptal.net/):
<details>
<summary>Code</summary>
<br />

```
import React from 'react'
import { Page, Menu, Link } from '@toptal/picasso'
import type { ReactNode } from 'react'
import { Link as RouterLink, BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import type { MenuItemProps } from '@toptal/picasso'

const Example = () => (
  <Page hamburgerId='hamburger-default-example'>
    <Page.TopBar
      variant='grey'
      centerContent={
        <Page.TopBar.Menu>
          <Page.TopBar.Item selected>Item 1</Page.TopBar.Item>
          <Page.TopBar.Item>Item 2</Page.TopBar.Item>
          <Page.TopBar.Item>Item 3</Page.TopBar.Item>
          <Page.TopBar.Item>Item 4</Page.TopBar.Item>
          <Page.TopBar.Item>Item 5</Page.TopBar.Item>
          <Page.TopBar.Item>Item 6</Page.TopBar.Item>
          {/* This won't be rendered. Maximum of 6 items is allowed */}
          <Page.TopBar.Item>Item 7</Page.TopBar.Item>
        </Page.TopBar.Menu>
      }
      rightContent={
        <Page.TopBarMenu
          name='Jacqueline Roque'
          avatar='./jacqueline-with-flowers-1954-square.jpg'
        >
         <Menu data-testid='talentDropdownMenu'>
        <Menu.Item
          to={''}
          disabled={false}
          data-testid='profileSettingsMenuLink'
        >
          Profile Settings
        </Menu.Item>
       <Menu.Item
          to={''}
          disabled={false}
          data-testid='profileSettingsMenuLink'
        >
            Legal Settings
        </Menu.Item>
        <Menu.Item
          to='/full-history'
          disabled={false}
          data-testid='fullHistoryMenuLink'
        >
          Full History
        </Menu.Item>

        <div style={{ borderBottom: '1px solid rgb(243, 244, 246)' }} />

        <Menu.Item
          to='/code-of-conduct'
          disabled={false}
          data-testid='codeOfConductMenuLink'
        >
          Code of Conduct
        </Menu.Item>
        <Menu.Item
          to='/terms-of-service'
          disabled={false}
          data-testid='portalAgreementMenuLink'
        >
          Terms of Service
        </Menu.Item>
        <Menu.Item
          to={''}
          disabled={false}
          data-testid='privacyPolicyMenuLink'
        >
          Privacy Policy
        </Menu.Item>

        <div style={{ borderBottom: '1px solid rgb(243, 244, 246)' }} />

        <Menu.Item onClick={() => {}}>Log Out</Menu.Item>
      </Menu>
        </Page.TopBarMenu>
      }
    />
  </Page>
)

export default Example

```
</details>

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| ![image](https://github.com/user-attachments/assets/fc97ab2a-99e6-4f6a-9b9f-5741ff04e0b2) | ![image](https://github.com/user-attachments/assets/0f137e04-259f-4095-851b-27e7cd13ec8c) |

### Development checks

- ~Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core~
- ~Annotate all `props` in component with documentation~
- ~Create `examples` for component~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- ~Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)~

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
4. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
4. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-5602]: https://toptal-core.atlassian.net/browse/FX-5602?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
